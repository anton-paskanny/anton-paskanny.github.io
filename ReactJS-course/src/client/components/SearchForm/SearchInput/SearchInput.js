import React from 'react';
import styled from 'styled-components';

const SearchInput = ({ className, value, handleInputChange }) => (
  <input name="movie"
        className={className}
        value={value}
        onChange={handleInputChange}
  />
);

export default styled(SearchInput)`
  display: block;
  width: 100%;
  padding: 10px;
  font-size: 18px;
  color: #fff;
  background-color: var(--secondary-color);
  border: none;
  border-bottom: 3px solid var(--primary-color);
`;
