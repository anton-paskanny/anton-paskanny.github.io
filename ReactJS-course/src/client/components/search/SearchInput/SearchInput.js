import React from 'react';

const SearchInput = props => (
    <div className="search-panel__input-wrapper">
        <input name="movie"
               className="search-panel__input"
               value={props.value}
               onChange={props.handleInputChange}
        />
    </div>
)

export default SearchInput;