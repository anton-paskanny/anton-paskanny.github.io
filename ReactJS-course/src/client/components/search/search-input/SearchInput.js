import React from 'react';

const SearchInput = props => (
    <div className="search__input-wrapper">
        <input name="movie" value={props.value} onChange={props.handleInputChange}/>
    </div>
)

export default SearchInput;