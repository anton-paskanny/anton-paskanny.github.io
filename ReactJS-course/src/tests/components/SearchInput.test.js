import React from 'react';
import { shallow } from 'enzyme';

import SearchInput from '../../client/components/Search/SearchInput/SearchInput';

let component, props;

describe('Movie', () => {
    beforeEach(() => {
        props = {
            value: '',
            handleInputChange: jest.fn()
        }
        component = shallow(<SearchInput {...props} />);
    });
    it('render component correctly', () => {
        expect(component).toMatchSnapshot();
    });
    it('should invoke onChange handler', () => {
        component.find('input').simulate('change');
        expect(props.handleInputChange).toHaveBeenCalled();
    });
});



