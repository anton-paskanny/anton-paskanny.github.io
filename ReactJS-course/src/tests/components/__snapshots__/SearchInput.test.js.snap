// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie render component correctly 1`] = `
<div
  className="search-panel__input-wrapper"
>
  <input
    className="search-panel__input"
    name="movie"
    onChange={[MockFunction]}
    value=""
  />
</div>
`;
